frontend/javascripts/oxalis/model_initialization.js

```diff
@@ -650,5 +655,24 @@ function applyLayerState(stateByLayer: UrlStateByLayer) {
         }
       }
     }
+
+    if (layerState.meshInfo) {
+      const { meshFileName, meshes } = layerState.meshInfo;
+
+      Store.dispatch(updateCurrentMeshFileAction(effectiveLayerName, meshFileName));
```

Comment:
Good that you mention it!
Currently, there is no way to find out which mesh file was used to load a precomputed mesh, after the fact. There is a `  +fileName?: string,` property in the `IsosurfaceInformation` in the store, but it doesn't seem to be populated, ever.

A related problem is which mapping was activated when an ad-hoc mesh was generated. Currently, this information is also not saved in any way.

I see two main directions we could take
(1) Save the `meshFileName` and `mappingName` information on a per-mesh basis and include it in the url. This would further increase the url length (maximum length for Edge [seems to be](https://stackoverflow.com/a/417184) 2047 characters).
(2) Revisit whether meshes should be invalidated/removed/hidden in certain situations (e.g.: if the mesh file is changed or the mapping is changed). Currently, the UI in the Segments tab can be pretty misleading. For example it is not possible to compute a mesh for segment 4 in the over segmentation, change the mapping to agglomerate_view_80 and compute a mesh for segment 4, there. I feel like we should clarify what should happen in these situations and then derive the url scheme from that, maybe?

Happy to jump on a call

Reply:
Good points!

I think, we should explore direction (2) and see how we can handle these situations elegantly. However, I'd say this is better suited for a follow-up issue. Do you agree?

Approach (1) would probably work (and would be relatively simple, when ignoring URL limits), but the actual complexity (highlighted by (2) wouldn't be solved. So, it doesn't feel like the right way.

Reply:
> I think, we should explore direction (2) and see how we can handle these situations elegantly. However, I'd say this is better suited for a follow-up issue. Do you agree?

Yes, I agree. So would you say, we keep the format as proposed in this PR for now (although it does not correctly work in some situations)?

Since we don't have a method to evolve the json url format to support older formats, yet, I would like to try to keep changes to a minimum, but it also doesn't feel right to merge this with known bugs :thinking: